pallets/subtensor/src/lib.rs

```diff
@@ -268,6 +268,42 @@ pub mod pallet {
         /// Additional information about the subnet
         pub additional: Vec<u8>,
     }
+
+    /// Data structure for stake related jobs.
+    #[derive(Encode, Decode, TypeInfo, Clone, PartialEq, Eq, Debug)]
+    pub enum StakeJob<AccountId> {
+        /// Represents jobs for "add_stake" and "add_stake_limit" operation
+        AddStake {
+            /// Hotkey account
+            hotkey: AccountId,
+            /// Coldkey account
+            coldkey: AccountId,
+            /// Subnet ID
+            netuid: u16,
+            /// Tao to be staked
+            tao_staked: u64,
+            /// Operation fee
+            fee: u64,
+            /// Signals whether it's a limit stake
+            limit: bool,
+        },
+        /// Represents jobs for "remove_stake" and "remove_stake_limit" operation
+        RemoveStake {
+            /// Hotkey account
+            hotkey: AccountId,
+            /// Coldkey account
+            coldkey: AccountId,
+            /// Subnet ID
+            netuid: u16,
+            /// Operation fee
+            fee: u64,
+            /// Alpha value
+            alpha: u64,
+            /// Signals whether it's a limit stake
+            limit: bool,
+        },
+    }
```

Comment:
A nice addition in the future (not the scope of the PR), could be to have a `freeze_enum` like the `freeze_struct` @sam0x17, wdyt?